### PR TITLE
chore(npm): added common npm/node development files to .gitignore

### PR DESCRIPTION
**Pull request by fabianmoronzirfas**
These files should never go into version control. Taken from https://github.com/github/gitignore/blob/master/Node.gitignore

### CODE REVIEW

**Comment by fabianmoronzirfas** [comment]
as mentioned by @trych in #87

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,20 @@ examples/ui/debug
 includes/ui
 # test files
 test/download/data/
+
+# Dependency directories
+node_modules
```
Comment (trych) [inline .gitignore]: Should this not be `node_modules/`?

**Comment by trych** [comment]
While we are at it, I think ignoring `includes/ui` and `examples/ui/debug` is outdated, as these directories don't exist anymore?

**Comment by fabianmoronzirfas** [comment]
removed

**Comment by fabianmoronzirfas** [comment]
Can I merge this?

**Comment by b-g** [comment]
@fabianmoronzirfas 
Sure + merged! Please go always ahead with such no brainer/maintenance PR! :)

**Comment by fabianmoronzirfas** [comment]
> Please go always ahead with such no brainer/maintenance PR! :)  

Roger roger.

**Pull request merged**